src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -1214,10 +1251,12 @@ private async Task AttachToTarget(SessionId sessionId, CancellationToken token)
             // see https://github.com/mono/mono/issues/19549 for background
             if (sessions.Add(sessionId))
             {
+                string checkUncaughtExceptions = "throw \"pause_on_uncaught\";";
+                string checkCaughtExceptions = "try {throw \"pause_on_caught\";} catch {}";
                 await SendMonoCommand(sessionId, new MonoCommands("globalThis.dotnetDebugger = true"), token);
                 Result res = await SendCommand(sessionId,
                     "Page.addScriptToEvaluateOnNewDocument",
-                    JObject.FromObject(new { source = "globalThis.dotnetDebugger = true; delete navigator.constructor.prototype.webdriver" }),
+                    JObject.FromObject(new { source = $"globalThis.dotnetDebugger = true; delete navigator.constructor.prototype.webdriver; {checkUncaughtExceptions} {checkCaughtExceptions}" }),
```

Comment:
doing it here is only going to work on new documents, would we be out of sync in the attach case still?  does it work if you add it to the SendMonoCommand part above?

Reply:
I tested two cases: The page is already loaded and then I connect debugger, it passes here and I can get the information that is expected.
The page is reloaded with the debugger already connected, it also passes here and I can get the information.
I'll test adding it to the SendMonoCommand.

Reply:
It does not work if I add it to SendMonoCommand because it uses Runtime.Evaluate and Runtime.Evaluate does not call Debugger.Pause when an exception is threw.